### PR TITLE
[Nano] Revise docstring for `input_spec` param in tf `InferenceOptimizer.optimize`/`trace`/`quantize`, and revise related how-to

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Revise docstring for `input_spec` param in tf `InferenceOptimizer.optimize`/`trace`/`quantize`, and revise related how-to

### 1. Why the change?

To make the usage of `input_spec` param in tf `InferenceOptimizer.trace`/`quantize` more clear for users.

### 2. Summary of the change 
- [Revise docstring for `input_spec` param in tf `InferenceOptimizer.optimize`/`trace`/`quantize`](https://yuwentestdocs.readthedocs.io/en/nano-tf-inputspec-api/doc/PythonAPI/Nano/tensorflow.html#bigdl.nano.tf.keras.InferenceOptimizer)
- [Revise docstring for `input_spec` param in tf `Model/Sequential.trace`/`quantize`](https://yuwentestdocs.readthedocs.io/en/nano-tf-inputspec-api/doc/PythonAPI/Nano/tensorflow.html#bigdl.nano.tf.keras.Model)
- Revised related howto guides:
  - [How to accelerate a TensorFlow inference pipeline through ONNXRuntime](https://yuwentestdocs.readthedocs.io/en/nano-tf-inputspec-api/doc/Nano/Howto/Inference/TensorFlow/accelerate_tensorflow_inference_onnx.html)
  - [How to accelerate a TensorFlow inference pipeline through OpenVINO](https://yuwentestdocs.readthedocs.io/en/nano-tf-inputspec-api/doc/Nano/Howto/Inference/TensorFlow/accelerate_tensorflow_inference_openvino.html)
  - [How to save and load optimized OpenVINO model in TensorFlow](https://yuwentestdocs.readthedocs.io/en/nano-tf-inputspec-api/doc/Nano/Howto/Inference/TensorFlow/tensorflow_save_and_load_openvino.html)
- Other small fixes to related docstrings

### 4. How to test?
- [x] Document test

### CODE REVIEW

File: python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py [hunk 1]
```diff
@@ -314,8 +314,11 @@ def trace(model: Model,
         :param model: The Keras model to trace.
         :param accelerator: The accelerator to use, defaults to None meaning staying in Keras
                             backend. 'openvino' and 'onnxruntime' are supported for now.
-        :param input_spec: A (tuple or list of) tf.TensorSpec or numpy array defining the
-                           shape/dtype of the input.
+        :param input_spec: (optional) A (tuple or list of) ``tf.TensorSpec`` or numpy array
+                           defining the shape/dtype of the input. If ``accelerator='onnxruntime'``,
+                           ``input_spec`` is required. If ``accelerator='openvino'``,
+                           ``input_spec`` is only required when you have a custom Keras model.
+>>>>>>> 76154c3972 (Revise docstrings for inputspec)
```
Comment (MeouSker77) [inline python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py:321]: may remove this rebase tag

File: python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py [hunk 2]
```diff
@@ -314,8 +314,11 @@ def trace(model: Model,
         :param model: The Keras model to trace.
         :param accelerator: The accelerator to use, defaults to None meaning staying in Keras
                             backend. 'openvino' and 'onnxruntime' are supported for now.
-        :param input_spec: A (tuple or list of) tf.TensorSpec or numpy array defining the
-                           shape/dtype of the input.
+        :param input_spec: (optional) A (tuple or list of) ``tf.TensorSpec`` or numpy array
```
Comment (MeouSker77) [inline python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py:317]: we may remove numpy array here, and only support `tf.TensorSpec`

File: python/nano/tutorial/notebook/inference/tensorflow/accelerate_tensorflow_inference_onnx.ipynb [hunk 3]
```diff
@@ -1,89 +1,749 @@
...
-      "outputs": [],
       "execution_count": null,
       "metadata": {
         "nbsphinx": "hidden"
-      }
+      },
+      "outputs": [],
+      "source": [
+        "!pip install --pre --upgrade bigdl-nano[tensorflow,inference] # install the nightly-built version\n",
+        "!source bigdl-nano-init # set environment variables"
```
Comment (MeouSker77) [inline python/nano/tutorial/notebook/inference/tensorflow/accelerate_tensorflow_inference_onnx.ipynb:50]: we should remind users to `source bigdl-nano-init` before starting jupyter kernel, it's too late to source here

**Review by MeouSker77 — APPROVED** [review]
others LGTM

File: python/nano/tutorial/notebook/inference/tensorflow/accelerate_tensorflow_inference_onnx.ipynb (hunk 3 above)
Comment (Oscilloscope98) [reply python/nano/tutorial/notebook/inference/tensorflow/accelerate_tensorflow_inference_onnx.ipynb:50]: Yes, we have a reminder in the next markdown cell to reminds user to `source bigdl-nano-init` before the kernel is started, which will be visible when user open this notebook locally, from online preview tools, etc. :)

File: python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py (hunk 1 above)
Comment (Oscilloscope98) [reply python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py:321]: Have changed. Thanks for reminding.

File: python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py (hunk 2 above)
Comment (MeouSker77) [reply python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py:318]: > Or we should hide the usage of numpy array in the api doc and only suggest user to use tf.TensorSepc?

yes, we have no UT for numpy array, and there is no need to support two different methods.

Comment (Oscilloscope98) [reply python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py]: Have updated, please have a look again. Thank you!

**Review by MeouSker77 — APPROVED** [review]
LGTM

**Pull request merged**